### ISSUE TITLE
[RFE] Pull quota-backend-bytes from etcd server?

### ISSUE THREAD

**Issue by evgkrsk** [ahrtr/etcd-defrag#40]
Is it really necessary to set quota bytes as CLI arg? Can it be pulled from etcd server? Now we need to tune such CLI flag accordingly to every cluster, looks like bad work.

And thanks for really usefull tool!

**Comment by ahrtr** [comment]
> Is it really necessary to set quota bytes as CLI arg?

Good point. I thought about it in the first place. But unfortunately, etcdserver doesn't export such an API to client for now.

**Comment by evgkrsk** [comment]
Ok, but maybe we can read prometheus metric for that? Maybe it will be necessary to set metrics http URL for that (as CLI arg or envvar?).

We have much less variance in metric URLs (compared to variance of quotas) and pulling quota from metrics may be handy to dynamically adjust etcd-defrag work after quota increase, for example.

Of course, such pull must be directly enabled to keep backward compability with current default quota in etcd-defrag.

**Comment by ahrtr** [comment]
> Ok, but maybe we can read prometheus metric for that?

It's workaround instead of graceful solution to me. I suggest to keep it as it's for now until etcdserver exports an API to support this.

**Comment by ahrtr** [comment]
FYI. https://github.com/etcd-io/etcd/pull/17877

**Comment by evs-ops** [comment]
Will it be implemented any time soon?

**Comment by ahrtr** [comment]
> Will it be implemented any time soon?

https://github.com/etcd-io/etcd/pull/17877 has already been merged, but only in 3.6.
